ibis/expr/scope.py

```diff
@@ -48,14 +48,8 @@ class Scope:
     def __init__(self, items: Dict[str, ScopeItem] = None):
         self._items = items or {}
 
-    def items(self):
-        return iter(self._items)
-
-    def __getitem__(self, op: Node) -> ScopeItem:
-        return self._items[op]
-
-    def __setitem__(self, op: Node, value: Any) -> None:
-        self._items[op] = value
+    def __contains__(self, op):
+        return op in self._items.keys()
```

Comment:
why not use 

```
return op in self._items?
```

Reply:
Oh I see Jeff's comments. Ok this is fine.

Reply:
hmm, no i agree this should be in `self._items`, only *iteration* should use the keys

Reply:
For my knowledge, is `k in self._items.keys` O(n) or O(1)? The type of self._items.keys is `dict_keys` so it's not clear to me if the `__contains__` method of `dict_key` is linear or constant.

Reply:
I looked around and seems that in Python 3.x, `k in self._items` and `k in self._items.keys()` are equivalent and they are all `O(1)`.

Reply:
I tested with a Python 3.6 kernel
```
>>> dic = dict.fromkeys(range(10**5))
>>> %timeit 10000 in dic
35.8 ns ± 0.757 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
>>> %timeit 10000 in dic.keys()
74.7 ns ± 0.385 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
seems `keys()` is still 2x slower than testing `in` for dict directly